packages/ra-core/src/controller/input/useReferenceInputController.ts

```diff
@@ -121,7 +122,10 @@ export const useReferenceInputController = (
         loaded: possibleValuesLoaded,
         loading: possibleValuesLoading,
         error: possibleValuesError,
-    } = useGetList(reference, pagination, sort, filterValues);
+    } = useGetList(reference, pagination, sort, filterValues, {
+        action: 'CUSTOM_QUERY',
+        enabled: isEnabled && isEnabled(filterValues),
```

Comment:
This won't work when isEnabled isn't set

Reply:
I actually relied on the `enabled` option default value being `true`. However, I can make it `true` explicitly if you think this is clearer.